lib/score_card.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class ScoreCard
+  attr_reader :frames
+
+  def initialize
+    @frames = []
+    @score = 0
+  end
+
+  def play(user_input)
+    final_frame ? tenth_round(user_input) : roll(user_input)
+  end
+
+  def roll(user_input)
+    if new_frame
+      frame = Frame.new
+      @frames << frame
+    else
+      frame = @frames.last
+    end
+    frame.roll(user_input)
+    total
+  end
+
+  def total
+    unless final_frame
+      @score = @frames.map.with_index do |frame, index|
+        bonus = 0
+        bonus += frame.rolls.first if previous_two_frames_are_strikes(index)
+        if previous_frame_is_strike(index)
+          bonus += frame.count
+        elsif previous_frame_is_spare(index)
+          bonus += frame.rolls.first
+        end
+        bonus + frame.count
```

Comment:
Ah, okay, now I understand why you needed to define it as a separate method! Disregard my earlier question. :)